pkg/virt-controller/watch/pool_test.go

```diff
@@ -244,23 +244,41 @@ var _ = Describe("Pool", func() {
 			})
 		}
 
+		expectVMCreation := func(nameMatcher types.GomegaMatcher) {
+			client.Fake.PrependReactor("create", "virtualmachines", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				created, ok := action.(testing.CreateAction)
+				Expect(ok).To(BeTrue())
+				createObj := created.GetObject().(*v1.VirtualMachine)
+				Expect(createObj.Name).To(nameMatcher)
+				Expect(createObj.GenerateName).To(Equal(""))
+				return true, created.GetObject(), nil
+			})
+		}
+
+		expectVMUpdate := func(revisionName string) {
+			client.Fake.PrependReactor("update", "virtualmachines", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+				created, ok := action.(testing.UpdateAction)
+				Expect(ok).To(BeTrue())
+				updateObj := created.GetObject().(*v1.VirtualMachine)
+				Expect(updateObj.Labels).To(HaveKeyWithValue(v1.VirtualMachinePoolRevisionName, revisionName))
+				return true, created.GetObject(), nil
+			})
+		}
+
 		It("should create missing VMs", func() {
-			pool, vm := DefaultPool(3)
+			pool, _ := DefaultPool(3)
 
 			addPool(pool)
 
 			poolRevision := createPoolRevision(pool)
 			expectControllerRevisionCreation(poolRevision)
-
-			vmInterface.EXPECT().Create(context.Background(), gomock.Any(), metav1.CreateOptions{}).Times(3).Do(func(ctx context.Context, arg interface{}, options metav1.CreateOptions) {
-				Expect(arg.(*v1.VirtualMachine).ObjectMeta.Name).To(HavePrefix(fmt.Sprintf("%s-", pool.Name)))
-				Expect(arg.(*v1.VirtualMachine).ObjectMeta.GenerateName).To(Equal(""))
-			}).Return(vm, nil)
+			expectVMCreation(HavePrefix(fmt.Sprintf("%s-", pool.Name)))
```

Comment:
Why don't we just query the client? Isn't Reactor overkill?

Reply:
I tried to stay aligned with the test style.
Here all the tests use reactors for both k8sclient and virt client... and I didn't want to refactor everything here.
We can do it in a followup PR, but I still don't have a stong opinion here actually. WDYT?

Reply:
Follow-up sounds good